src/drv.cc

```diff
@@ -63,6 +63,19 @@ auto queryCacheStatus(nix::Store &store,
     }
     return Drv::CacheStatus::NotBuilt;
 };
+
+auto outputs_to_json(const std::map<std::string, std::optional<std::string>>
+                         &outputs) -> nlohmann::json {
+    nlohmann::json json;
+    for (const auto &pair : outputs) {
+        if (pair.second) {
+            json[pair.first] = pair.second.value();
+        } else {
+            json[pair.first] = nullptr;
+        }
+    }
+    return json;
+}
```

Comment:
I think with the right headers you shouldn't need this function?

Reply:
https://github.com/NixOS/nix/blob/3180c09723c3b61d682732c308c74e2087ac628f/src/libutil/json-utils.hh#L43-L124 you just need this stuff in scope

Reply:
https://github.com/nlohmann/json/issues/1749#issuecomment-2498334192 oh huh, the upstream issue finally saw some progress too, and I left this comment.

Reply:
Solved.